File: 230420/累積和/AGC023-A.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

int main() {
    int n;
    cin >> n;
    vector<long long> a(n);
    for (int i = 0; i < n; i++) cin >> a[i];
    
    vector<long long> s(n + 1, 0);
    for (int i = 0; i < n; i++) {
        s[i + 1] = s[i] + a[i];
    }
    
    long long ans = 0;
    map<long long, long long> counter;
    for (int i = 0; i < n + 1; i++) {
        ans += counter[s[i]];
        counter[s[i]]++;
    }
    
    cout << ans << endl;
    return 0;
}

File: 230420/累積和/AGC023-A.py
n = int(input())
a = list(map(int, input().split()))

s = [0 for i in range(n + 1)]
for i in range(n):
    s[i + 1] = s[i] + a[i]

ans = 0
counter = dict()
for i in range(n + 1):
    if s[i] not in counter:
        counter[s[i]] = 0
    ans += counter[s[i]]
    counter[s[i]] += 1

print(ans)